src/parse-nm.c

```diff
@@ -64,17 +65,306 @@ ap_type_from_str(const char* type_str)
+        _kf_clear_key(kf, group, "never-default");
+    }
+    handle_generic_uint(kf, group, "route-metric", &(*dataptr).metric, NETPLAN_METRIC_UNSPEC);
+}
+
+static void
+parse_search_domains(GKeyFile* kf, const gchar* group, GArray** domains_arr)
+{
+    g_assert(domains_arr);
+    gsize len = 0;
+    gchar **split = g_key_file_get_string_list(kf, group, "dns-search", &len, NULL);
+    if (split) {
+        if (len == 0) {
+            _kf_clear_key(kf, group, "dns-search");
+            return;
+        }
+        if (!*domains_arr)
+            *domains_arr = g_array_new(FALSE, FALSE, sizeof(char*));
+        for(unsigned i = 0; split[i]; ++i) {
+            char* s = g_strdup(split[i]); //no need to free, will stay in netdef
+            g_array_append_val(*domains_arr, s);
+        }
+        _kf_clear_key(kf, group, "dns-search");
+        g_strfreev(split);
+    }
+}
+
+static void
+parse_nameservers(GKeyFile* kf, const gchar* group, GArray** nameserver_arr)
+{
+    g_assert(nameserver_arr);
+    gchar **split = g_key_file_get_string_list(kf, group, "dns", NULL, NULL);
+    if (split) {
+        if (!*nameserver_arr)
+            *nameserver_arr = g_array_new(FALSE, FALSE, sizeof(char*));
+        for(unsigned i = 0; split[i]; ++i) {
+            if (strlen(split[i]) > 0) {
+                gchar* s = g_strdup(split[i]); //no need to free, will stay in netdef
+                g_array_append_val(*nameserver_arr, s);
+            }
+        }
+        _kf_clear_key(kf, group, "dns");
+        g_strfreev(split);
+    }
+}
+
+static void
+parse_dot1x_auth(GKeyFile* kf, NetplanAuthenticationSettings* auth)
+{
+    g_assert(auth);
+    g_autofree gchar* method = g_key_file_get_string(kf, "802-1x", "eap", NULL);
+
+    if (method && g_strcmp0(method, "tls") == 0) {
+        auth->eap_method = NETPLAN_AUTH_EAP_TLS;
+        _kf_clear_key(kf, "802-1x", "eap");
+    } else if (method && g_strcmp0(method, "peap") == 0) {
+        auth->eap_method = NETPLAN_AUTH_EAP_PEAP;
+        _kf_clear_key(kf, "802-1x", "eap");
+    } else if (method && g_strcmp0(method, "ttls") == 0) {
+        auth->eap_method = NETPLAN_AUTH_EAP_TTLS;
+        _kf_clear_key(kf, "802-1x", "eap");
+    }
+
+    handle_generic_str(kf, "802-1x", "identity", &auth->identity);
+    handle_generic_str(kf, "802-1x", "anonymous-identity", &auth->anonymous_identity);
+    if (!auth->password)
+        handle_generic_str(kf, "802-1x", "password", &auth->password);
+    handle_generic_str(kf, "802-1x", "ca-cert", &auth->ca_certificate);
+    handle_generic_str(kf, "802-1x", "client-cert", &auth->client_certificate);
+    handle_generic_str(kf, "802-1x", "private-key", &auth->client_key);
+    handle_generic_str(kf, "802-1x", "private-key-password", &auth->client_key_password);
+    handle_generic_str(kf, "802-1x", "phase2-auth", &auth->phase2_auth);
+}
+
+static void
+parse_bond_arp_ip_targets(GKeyFile* kf, GArray **targets_arr)
+{
```

Comment:
```suggestion
{
    g_assert(targets_arr);
```

Also for consistency, I suppose.